libs/wingc/src/type_check.rs

```diff
@@ -1228,12 +1265,37 @@ impl Types {
 			.and_then(|t| t.as_ref().map(|t| t.type_))
 	}
 
+	/// Obtain the type of a given expression node. Will panic if the expression has not been type checked yet.
+	pub fn get_expr_type_after_check(&self, expr: &Expr) -> TypeRef {
+		self
+			.type_for_expr
+			.get(expr.id)
+			.and_then(|t| t.as_ref().map(|t| t.type_))
+			.expect("All expressions should have a type")
+	}
```

Comment:
It could be worth just changing `get_expr_type` to return the unwrapped option, I'm not sure there's a need for the two method variants

Reply:
I went ahead and changed it now, after checking around it looks like we just unwrap it everywhere.